staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -608,211 +621,50 @@ See http://k8s.io/docs/reference/using-api/api-concepts/#conflicts`, err)
+		o.visitedNamespaces.Insert(info.Namespace)
 	}
-
-	for _, resource := range *pruneResources {
-		addedMapping, err := mapper.RESTMapping(schema.GroupKind{Group: resource.group, Kind: resource.kind}, resource.version)
-		if err != nil {
-			return nil, nil, fmt.Errorf("invalid resource %v: %v", resource, err)
-		}
-		if resource.namespaced {
-			namespaced = append(namespaced, addedMapping)
-		} else {
-			nonNamespaced = append(nonNamespaced, addedMapping)
-		}
-	}
-
-	return namespaced, nonNamespaced, nil
 }
 
-type pruner struct {
-	mapper        meta.RESTMapper
-	dynamicClient dynamic.Interface
-
-	visitedUids   sets.String
-	labelSelector string
-	fieldSelector string
-
-	cascade      bool
-	serverDryRun bool
-	dryRun       bool
-	gracePeriod  int
-
-	toPrinter func(string) (printers.ResourcePrinter, error)
-
-	out io.Writer
-}
-
-func (p *pruner) prune(namespace string, mapping *meta.RESTMapping) error {
-	objList, err := p.dynamicClient.Resource(mapping.Resource).
-		Namespace(namespace).
-		List(metav1.ListOptions{
-			LabelSelector: p.labelSelector,
-			FieldSelector: p.fieldSelector,
-		})
-	if err != nil {
-		return err
-	}
-
-	objs, err := meta.ExtractList(objList)
+// MarkNamespaceVisited keeps track of UIDs of the applied
+// objects. Used for pruning.
+func (o *ApplyOptions) MarkObjectVisited(info *resource.Info) error {
+	metadata, err := meta.Accessor(info.Object)
 	if err != nil {
 		return err
 	}
-
-	for _, obj := range objs {
-		metadata, err := meta.Accessor(obj)
-		if err != nil {
-			return err
-		}
-		annots := metadata.GetAnnotations()
-		if _, ok := annots[corev1.LastAppliedConfigAnnotation]; !ok {
-			// don't prune resources not created with apply
-			continue
-		}
-		uid := metadata.GetUID()
-		if p.visitedUids.Has(string(uid)) {
-			continue
-		}
-		name := metadata.GetName()
-		if !p.dryRun {
-			if err := p.delete(namespace, name, mapping); err != nil {
-				return err
-			}
-		}
-
-		printer, err := p.toPrinter("pruned")
-		if err != nil {
-			return err
-		}
-		printer.PrintObj(obj, p.out)
-	}
+	o.visitedUids.Insert(string(metadata.GetUID()))
 	return nil
 }
 
-func (p *pruner) delete(namespace, name string, mapping *meta.RESTMapping) error {
-	return runDelete(namespace, name, mapping, p.dynamicClient, p.cascade, p.gracePeriod, p.serverDryRun)
-}
-
-func runDelete(namespace, name string, mapping *meta.RESTMapping, c dynamic.Interface, cascade bool, gracePeriod int, serverDryRun bool) error {
-	options := &metav1.DeleteOptions{}
-	if gracePeriod >= 0 {
-		options = metav1.NewDeleteOptions(int64(gracePeriod))
-	}
-	if serverDryRun {
-		options.DryRun = []string{metav1.DryRunAll}
-	}
-	policy := metav1.DeletePropagationForeground
-	if !cascade {
-		policy = metav1.DeletePropagationOrphan
-	}
-	options.PropagationPolicy = &policy
-	return c.Resource(mapping.Resource).Namespace(namespace).Delete(name, options)
+func (o *ApplyOptions) SetPreProcessorFn(pre func(*ApplyOptions) error) {
+	o.preprocessFn = pre
```

Comment:
We should be consistent in the naming here. Now the function is capitalizing `Processor` while the field does not.